src/actions.ts

```diff
@@ -341,16 +341,23 @@ export const goToLineBoundary = (
 export const navigateLine = (
   editor: Editor,
   selection: EditorSelection,
-  direction: 'up' | 'down',
+  position: 'next' | 'prev' | 'top' | 'bottom',
 ) => {
   const pos = selection.head;
   let line: number;
 
-  if (direction === 'up') {
+  if (position === 'prev') {
     line = Math.max(pos.line - 1, 0);
-  } else {
+  }
+  if (position === 'next') {
     line = Math.min(pos.line + 1, editor.lineCount() - 1);
   }
+  if (position === 'top') {
+    line = 0;
+  }
+  if (position === 'bottom') {
+    line = editor.lineCount() - 1;
```

Comment:
One thing to clarify about behaviour: when moving to the first/last line, do we care about the character position as well?

I'm not sure how it behaves on mobile (I don't have the iOS app), but I'm thinking of how `⌘`+`↑` and `⌘`+`↓` on Mac navigate to the very first and very last character.

Happy to go with either option.

Reply:
I'm not particular about this, but I thought that if combining it with "Go to start/end of line" would do the same thing, and some people might be happier if the command cursor position was preserved.

However, I see that "Go to first/last line" on iOS also moves to the start/end of the line, as do `⌘`+`↑` and `⌘`+`↓` on macOS.
It seems less confusing to keep it in line with Obsidian's native behavior, so I'll do that.

Reply:
Fixed